docs/src/pages/components/data-grid/group-pivot/group-pivot.md

```diff
@@ -6,13 +6,111 @@ title: Data Grid - Group & Pivot
+  { path: ['Sarah', 'Thomas', 'Karen'], jobTitle: 'Sales Person', id: 3 },
+];
+
+<DataGridPro
+  treeData
+  getTreeDataPath={(row) => row.path}
+  rows={rows}
+  columns={columns}
+/>;
+
+// With transformation
+const rows: GridRowsProp = [
+  { path: 'Sarah', jobTitle: 'CEO', id: 0 },
+  { path: 'Sarah/Thomas', jobTitle: 'Head of Sales', id: 1 },
+  { path: 'Sarah/Thomas/Robert', jobTitle: 'Sales Person', id: 2 },
+  { path: 'Sarah/Thomas/Karen', jobTitle: 'Sales Person', id: 3 },
+];
+
+<DataGridPro
+  treeData
+  getTreeDataPath={(row) => row.path.split('/')}
+  rows={rows}
+  columns={columns}
+/>;
+```
+
+{{"demo": "pages/components/data-grid/group-pivot/BasicTreeData.js", "bg": "inline", "defaultCodeOpen": false}}
+
+### Custom grouping column
+
+Use the `groupingColDef` prop to customize the rendering of the grouping column.
+
+{{"demo": "pages/components/data-grid/group-pivot/CustomGroupingColumnTreeData.js", "bg": "inline", "defaultCodeOpen": false}}
+
+### Group expansion
+
+Use the `defaultGroupingExpansionDepth` prop to expand all the groups up to a given depth when loading the data.
+If you want to expand the whole tree, set `defaultGroupingExpansionDepth = -1`
+
+{{"demo": "pages/components/data-grid/group-pivot/DefaultGroupingExpansionDepthTreeData.js", "bg": "inline", "defaultCodeOpen": false}}
+
+Use the `setRowChildrenExpansion` method on `apiRef` to programmatically set the expansion of a row.
+
+{{"demo": "pages/components/data-grid/group-pivot/SetRowExpansionTreeData.js", "bg": "inline", "defaultCodeOpen": false}}
+
+### Gaps in the tree
+
+If some entries are missing to build the full tree, the `DataGridPro` will automatically create rows to fill those gaps.
+
+{{"demo": "pages/components/data-grid/group-pivot/TreeDataWithGap.js", "bg": "inline", "defaultCodeOpen": false}}
+
+### Filtering
+
+A node is included if one of the following criteria is met:
+
+- at least one of its descendant is passing the filters
+- it is passing the filters
+
+By default, the filtering is applied to every depth of the tree.
+You can limit the filtering to the top level rows with the `disableChildrenFiltering` prop.
+
+{{"demo": "pages/components/data-grid/group-pivot/DisableChildrenFilteringTreeData.js", "bg": "inline", "defaultCodeOpen": false}}
+
+### Sorting
+
+By default, the sorting is applied to every depth of the tree.
+You can limit the sorting to the top level rows with the `disableChildrenSorting` prop.
+
+{{"demo": "pages/components/data-grid/group-pivot/DisableChildrenSortingTreeData.js", "bg": "inline", "defaultCodeOpen": false}}
+
+> If you are using `sortingMode="server"`, you need to always set the children of a row after this row.
+> For instance:
+
+```ts
+// The row A.A is immediately after its parent
+const validRows = [{ path: ['A'] }, { path: ['A', 'A'] }, { path: ['B'] }];
+
+// The row A.A is not immediately after its parent
+const invalidRows = [{ path: ['A'] }, { path: ['B'] }, { path: ['A', 'A'] }];
```

Comment:
Add emojis to make it clear what is the wrong pattern. Two examples: https://mui.com/guides/minimizing-bundle-size/#option-1 and https://mui.com/customization/how-to-customize/#main-content